File: adduo.restoudaobra.website/wwwroot/_script/spa/ad/adService.js
ROApp.service('adService', ['$http', 'adFactory', function ($http, adFactory) {

    this.save = function (product) {

        var request = adFactory.helper.createRequest(product);

        return $http.post('/api/ad/register', request);
    }

    this.getGuid = function () {

        return $http.get('/api/ad/getguid');
    }

    this.list = function () {
        return $http.get('/api/ad/list');
    }

    this.get = function (guid) {
        return $http.get('/api/ad/get?guid=' + guid);
    }

    this.detail = function (guid) {
        return $http.get('/api/ad/detail?guid=' + guid);
    }


    this.changeStatus = function (dto) {
        var request = adFactory.helper.createRequest(dto);
        return $http.post('/api/ad/changestatus', request);
    }


}]);

File: adduo.restoudaobra.service/ad/AdService.cs
using adduo.basetype.envelope;
using adduo.helper.property;
using adduo.restoudaobra.constants;
using adduo.restoudaobra.dto;
using adduo.restoudaobra.ie.dal;
using adduo.restoudaobra.ie.service;

namespace adduo.restoudaobra.service.ad
{
    public class AdService : BaseService, IAdService
    {
        private IAdDAL productDAL { get; set; }

        public AdService(IAdDAL productDAL)
        {
            this.productDAL = productDAL;
        }

        public BaseResponse<AdRegisterDTO> Save(BaseRequest<AdRegisterDTO> request)
        {
            if (request.Success)
            {
                productDAL.Save(request.Dto);
            }

            return request.CreateResponse();
        }

        public void ChangeStatus(BaseRequest<AdChangeStatusDTO> request)
        {
            var dto = request.Dto;

            productDAL.ChangeStatus(dto);

        }

        public void Validation(BaseRequest<AdRegisterDTO> request)
        {
            PropertyHelper.ResetProperty(request.Dto);

            if (request.Dto.Type == AD_TYPE.SALE)
            {
                request.Dto.AddPropertyToValidation(request.Dto.Price);
            }

            foreach (var property in request.Dto.Properties)
            {
                property.Validate();
            }


            request.Success = PropertyHelper.ReturnTrueIfAllValid(request.Dto);
        }

    }
}
